code/game/objects/structures/watercloset.dm

```diff
@@ -472,3 +466,31 @@
 	icon_state = "bounty-open"
 	color = null
 	alpha = 255
+
+/obj/structure/curtain/proc/toggle(mob/M)
+    if (check(M))
+        open = !open
+        playsound(loc, 'sound/effects/curtain.ogg', 50, 1)
+        update_appearance()
+    return
+
+/obj/structure/curtain/proc/check(mob/M)
+    if (istype(src, /obj/structure/curtain/directional))
+        if (src.dir != get_dir(src, M))
```

Comment:
I think this will dislike diagonal states. Try this, perhaps.
```suggestion
        if (get_dir(src, M) & dir)
```
if direction is NORTH, then NORTH | EAST & NORTH will return NORTH which will pass the truthy test.
If the player is SOUTH | EAST from the curtain, then (SOUTH | EAST) & NORTH will return 0 which will fail the truthy test.

Reply:
& is a bitwise operator which compares bits. Because directional states are bit flags (NORTH: 1, South:2, East: 4, West: 8) then we get the binary being this:
NORTH: 0001
SOUTH: 0010
EAST: 0100
WEST: 1000
& allows us to check if certain bits are activated. If we are coming from the north west, then we have the binary `1001` which has the north flag in it, so should allow us to use the curtain.
1001
0001 & (1 if 1 on both, 0 otherwise)
-----
0001

Reply:
this is a much better solution, thank you.